### PR TITLE
[v1.x] Onnx export support for batch_dot

### PR DESCRIPTION
**Pull request by waytrue17**
## Description ##
Onnx export support for batch_dot

### Essentials ###
- [ ] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py [hunk 1]
```diff
@@ -3275,3 +3275,78 @@ def convert_gather_nd(node, **kwargs):
     ]
 
     return nodes
+
+
+@mx_op.register("batch_dot")
+def convert_batch_dot(node, **kwargs):
+    """Map MXNet's gather_ND operator attributes to onnx's operator.
```
Comment (josephevans) [inline python/mxnet/contrib/onnx/mx2onnx/_op_translations.py:3282]: comment is wrong ;)

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py [hunk 2]
```diff
@@ -3275,3 +3275,78 @@ def convert_gather_nd(node, **kwargs):
...
+    """
+    from onnx.helper import make_node
+    name, input_nodes, attrs = get_inputs(node, kwargs)
+
+    lhs = input_nodes[0]
+    rhs = input_nodes[1]
+    input_type = kwargs['in_type']
+    dtype = onnx.mapping.TENSOR_TYPE_TO_NP_TYPE[input_type]
+    transpose_a = str(attrs.get('transpose_a', '0'))
```
Comment (josephevans) [inline python/mxnet/contrib/onnx/mx2onnx/_op_translations.py:3291]: Would it make sense to set the defaults to these to 'False', to simplify where you check transpose_[ab] below?

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py [hunk 3]
```diff
@@ -3275,3 +3275,78 @@ def convert_gather_nd(node, **kwargs):
...
+
+    if transpose_a in ['0', 'False']:
+        nodes += [
+            make_node('Add', [lhs, name+'_0f'], [name+'_lhs']),
+        ]
+    else:
+        nodes += [
+            make_node('Shape', [lhs], [name+'_lhs_shape']),
+            make_node('Shape', [name+'_lhs_shape'], [name+'_lhs_dim']),
+            # make_node('Sub', [name+'_lhs_dim', name+'_2'], [name+'_lhs_sub']),
```
Comment (josephevans) [inline python/mxnet/contrib/onnx/mx2onnx/_op_translations.py:3312]: remove commented out line?

**Review by josephevans — COMMENTED** [review]
Just a few small issues.

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py (hunk 1 above)
Comment (waytrue17) [reply python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: Good catch:)

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py (hunk 2 above)
Comment (waytrue17) [reply python/mxnet/contrib/onnx/mx2onnx/_op_translations.py:3291]: Fixed

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py (hunk 3 above)
Comment (waytrue17) [reply python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: Removed

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py [hunk 4]
```diff
@@ -3275,3 +3275,77 @@ def convert_gather_nd(node, **kwargs):
...
+        create_tensor([-1], name+'_-1', kwargs['initializer']),
+        create_tensor([0], name+'_0', kwargs['initializer']),
+        create_tensor([2], name+'_2', kwargs['initializer']),
+        create_tensor([100], name+'_100', kwargs['initializer']),
+        create_tensor([0], name+'_0f', kwargs['initializer'], dtype=dtype),
+    ]
+
+    if transpose_a == 'False':
+        nodes += [
+            make_node('Add', [lhs, name+'_0f'], [name+'_lhs']),
```
Comment (Zha0q1) [inline python/mxnet/contrib/onnx/mx2onnx/_op_translations.py:3306]: can we get rid of this add by writing four `make_node('MatMul', [name+'_lhs', name+'_rhs'], [name])` in each case? this drags performance down

Comment (Zha0q1) [inline python/mxnet/contrib/onnx/mx2onnx/_op_translations.py:3306]: can we get rid of this add by writing four make_node('MatMul', [name+'_lhs', name+'_rhs'], [name]) in each case? this drags performance down

**Review by Zha0q1 — CHANGES_REQUESTED** [review]

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py (hunk 4 above)
Comment (waytrue17) [reply python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: Addressed

**Comment by waytrue17** [comment]
@mxnet-bot run ci [unix-gpu]

**Review by Zha0q1 — APPROVED** [review]
LGTM!

**Pull request merged**